src/idl_parser.cpp

```diff
@@ -2530,6 +2530,12 @@ std::set<std::string> Parser::GetIncludedFilesRecursive(
 
 // Schema serialization functionality:
 
+static std::string UnqualifiedName(std::string name) {
```

Comment:
This could fail badly in case the filename does not have a dot, but the foldername has?